src/main/java/com/epam/aidial/core/log/GfLogStore.java

```diff
@@ -199,4 +216,108 @@ private static String formatTimestamp(long timestamp) {
+                    usage = tree.get("usage");
+                }
+                if (tree.get("statistics") != null) {
+                    statistics = tree.get("statistics");
+                }
+                if (tree.get("system_fingerprint") != null) {
+                    systemFingerprint = tree.get("system_fingerprint");
+                }
+                if (model == null && tree.get("model") != null) {
+                    model = tree.get("model");
+                }
+                last = tree;
+                ArrayNode choices = (ArrayNode) tree.get("choices");
+                if (choices == null) {
+                    // skip error message
+                    continue;
+                }
+                JsonNode curChoice = choices.get(0);
+                choice.set("finish_reason", curChoice.get("finish_reason"));
+                JsonNode delta = curChoice.get("delta");
+                if (delta.get("custom_content") != null) {
+                    message.set("custom_content", delta.get("custom_content"));
+                }
+                if (delta.get("tool_calls") != null) {
+                    message.set("tool_calls", delta.get("tool_calls"));
+                }
+                if (delta.get("function_call") != null) {
+                    message.set("function_call", delta.get("function_call"));
+                }
+                JsonNode contentNode = delta.get("content");
+                if (contentNode != null) {
+                    content.append(contentNode.textValue());
+                }
+            }
+
+            if (last == null) {
+                log.warn("no chunk is found in streaming response");
+                return "{}";
+            }
+
+            result.set("id", last.get("id"));
+            result.put("object", "chat.completion");
+            result.set("created", last.get("created"));
+            result.set("model", model);
+
+            if (usage != null) {
+                result.set("usage", usage);
+            }
+            if (statistics != null) {
+                result.set("statistics", statistics);
+            }
+            if (systemFingerprint != null) {
+                result.set("system_fingerprint", systemFingerprint);
+            }
+
+            if (content.isEmpty()) {
+                // error
+                return ProxyUtil.convertToString(result);
+            }
+
+            ArrayNode choices = ProxyUtil.MAPPER.createArrayNode();
+            result.set("choices", choices);
+            choices.add(choice);
+            choice.put("index", 0);
+            message.put("role", "assistant");
+            message.put("content", content.toString());
+
+            return ProxyUtil.convertToString(result);
+        } catch (Throwable e) {
+            log.warn("Can't assemble streaming response", e);
+            return "{}";
+        }
+    }
+
+    static boolean isStreamingResponse(Buffer response) {
```

Comment:
please, add a comment with examples, hard to understand what is streaming and non-streaming format you are checking

Reply:
fixed.